### PR TITLE
feat(transfer_metadata): Added functions to transfer metadata and col…

### PR DESCRIPTION
**Pull request by MelchiorSchuh**
…lections from one model to another with mappings.

### CODE REVIEW

File: include/geode/mesh/core/grid.hpp [hunk 1]
```diff
@@ -175,7 +175,7 @@ namespace geode
                 { 2, 6 }, { 3, 7 } } };
 
         static constexpr std::array< std::array< geode::local_index_t, 4 >, 6 >
-            CELL_FACET_VERTICES_3D{ { { 0, 4, 6, 2 }, { 1, 5, 7, 3 },
+            CELL_FACET_VERTICES_3D{ { { 0, 2, 6, 4 }, { 1, 5, 7, 3 },
```
Comment (BotellaA) [inline include/geode/mesh/core/grid.hpp:178]: sure about this change?

File: src/geode/model/representation/core/detail/transfer_collections.cpp [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+namespace
+{
+    template < typename Model >
+    void transfer_corner_collections( const Model& old_model,
+        const Model& new_model,
+        typename Model::Builder& new_model_builder,
+        const geode::ModelGenericMapping& component_mapping )
+    {
+        if( !component_mapping.has_mapping_type(
+                geode::Corner3D::component_type_static() ) )
```
Comment (BotellaA) [inline src/geode/model/representation/core/detail/transfer_collections.cpp:49]:
```suggestion
                geode::Corner<Model::dim>::component_type_static() ) )
```
Maybe better like this? And same for the others?

File: src/geode/model/representation/core/detail/transfer_collections.cpp [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+            transfer_surface_collections(
+                old_brep, new_brep, new_brep_builder, component_mapping );
+            transfer_block_collections(
+                old_brep, new_brep, new_brep_builder, component_mapping );
+            if( !component_mapping.has_mapping_type(
+                    geode::Surface3D::component_type_static() ) )
+            {
+                return;
+            }
+            const auto& surfaces_mapping =
```
Comment (BotellaA) [inline src/geode/model/representation/core/detail/transfer_collections.cpp:276]: add also a function for this, even if it is used once, so the transfer_brep_collections() has only one level of abstraction

**Review by BotellaA — APPROVED** [review]

**Comment by BotellaA** [comment]
Could you use this helper in the Model copy/clone?

File: include/geode/mesh/core/grid.hpp (hunk 1 above)
Comment (MelchiorSchuh) [reply include/geode/mesh/core/grid.hpp:178]: Yes, this was a bug: the facet was not oriented like the others

**Pull request merged**